File: source/_posts/1.leetcode/27.移除元素/27.移除元素.py
#
# @lc app=leetcode.cn id=27 lang=python3
#
# [27] 移除元素
#
from typing import List


# @lc code=start
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        j = len(nums) - 1
        for i in range(len(nums) - 1, -1, -1):
            if nums[i] == val:
                nums[i], nums[j] = nums[j], nums[i]
                j -= 1
        return j + 1


# @lc code=end

File: source/_posts/Algorithm/1.leetcode/Easy/27.移除元素/27.移除元素.go
/*
 * @lc app=leetcode.cn id=27 lang=golang
 *
 * [27] 移除元素
 */
package main

import (
	"fmt"
)

// @lc code=start
func removeElement(nums []int, val int) int {
	if len(nums) == 0 {
		return 0
	}
	s := 0
	for i := 0; i < len(nums)-s; i++ {
		if nums[i] == val {
			s++
			// 相当于 delete num[j]
			for j := i; j < len(nums)-1; j++ {
				nums[j] = nums[j+1]
			}
			i--
		}
	}
	return len(nums) - s
}

// @lc code=end

func main() {
	list := []int{3, 2, 2, 3}
	fmt.Println(removeElement(list, 3))
}
